plugins/collections/get.js

```diff
@@ -7,42 +7,95 @@ const getSchema = schema.models.collection.get;
+            ))
+        ))
+        .then((prJobsBuilds) => {
+            // The number of elements is the number of prs open
+            result.prs.open = prJobsBuilds.length;
+
+            prJobsBuilds.forEach((jobBuilds) => {
+                // Check the first element (last build) if pr job failed
+                if (jobBuilds[0].status === 'FAILURE') {
+                    result.prs.failing += 1;
+                }
+            });
+
+            return result;
+        })
+        .catch(() => result);
+}
+
+/**
+ * Helper function to get last builds of pipeline
+ *
+ * @param   {Object}  pipeline      Pipeline to get last builds for
+ * @param   {Factory} eventFactory  Factory for getting last event model
+ * @returns {Object}
+ */
+function getPipelineHealth(pipeline, eventFactory) {
+    const result = Object.assign({}, pipeline);
 
-            if (!pipeline.lastEventId) {
+    if (!pipeline.lastEventId) {
+        return result;
+    }
+
+    return eventFactory.get(pipeline.lastEventId)
+        .then((event) => {
+            if (!event) {
                 return result;
             }
 
-            return eventFactory.get(pipeline.lastEventId)
-                .then((event) => {
-                    if (!event) {
-                        return result;
+            return event.getBuilds()
+                .then((builds) => {
+                    if (builds.length) {
+                        // The events are sorted by most recent first. Need to reverse the order
+                        // to allow for matching with workflow job on the UI
+                        result.lastBuilds = builds.map(b => b.toJson()).reverse();
                     }
 
-                    return event.getBuilds()
-                        .then((builds) => {
-                            if (builds.length) {
-                                // The events are sorted by most recent first. Need to reverse the order
-                                // to allow for matching with workflow job on the UI
-                                result.lastBuilds = builds.map(b => b.toJson()).reverse();
-                            }
+                    return result;
+                });
+        })
+        .catch(() => result);
+}
 
-                            return result;
-                        });
-                })
-                .catch(() => result);
-        });
+/**
+ * Helper function to populate pipelines with last builds and pr info
+ *
+ * @param   {Array}   pipelines    Array of pipeline models to get last builds for
+ * @param   {Factory} eventFactory Factory for getting last event model
+ * @returns {Array}
+ */
+function getPipelinesInfo(pipelines, eventFactory) {
+    return pipelines
+        // Filter out pipelines that don't exist
+        .filter(pipeline => !!pipeline)
+        .map(pipeline => (
+            getPipelinePRInfo(pipeline)
+                .then(pipelineWithPRInfo => getPipelineHealth(pipelineWithPRInfo, eventFactory))
```

Comment:
`getPipelinePRInfo` & `getPipelineHealth` can be called in parallel. And then be funneled to a `Promise.all` to combine them

Reply:
Fixed 👍